### PR TITLE
PlayerShotFade

### PR DESCRIPTION
**Pull request by Ashthecool**
The Player shots fade slowly when made
Made a new global variable PLAYER_SHOT_FADE

### CODE REVIEW

File: my_game.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ PLAYER_SPEED_LIMIT = 5  # px/update
 PLAYER_INVINCIBILITY_SECONDS = 5  # secs
 PLAYER_FIRE_RATE = 0.2  # secs
 PLAYER_SHOT_RANGE = 400  # px
+PLAYER_SHOT_FADE = -2 #alpha
```
Comment (tobiasbp) [inline my_game.toml:20]: Please make this a positive value. Then update line 221.

**Review by tobiasbp — CHANGES_REQUESTED** [review]

File: my_game.toml (hunk 1 above)
Comment (Ashthecool) [reply my_game.toml]: Done

File: my_game.py [hunk 2]
```diff
@@ -215,6 +217,9 @@ def update(self):
         self.center_x += self.change_x
         self.center_y += self.change_y
 
+        # Smooth fade
+        self.alpha += -CONFIG['PLAYER_SHOT_FADE']
```
Comment (tobiasbp) [inline my_game.py:221]: Can we have _-=_ ?

**Review by tobiasbp — CHANGES_REQUESTED** [review]

File: my_game.py (hunk 2 above)
Comment (Ashthecool) [reply my_game.py:221]: Yes, we can!

File: my_game.py [hunk 3]
```diff
@@ -218,6 +218,11 @@ def update(self):
         # wrap
         wrap(self)
 
+        # Checks when the distance of the shot is almost over
+        if self.distance_traveled > 300:
```
Comment (tobiasbp) [inline my_game.py:222]: No magic numbers. Make a config variable.

File: my_game.py [hunk 4]
```diff
@@ -218,6 +218,11 @@ def update(self):
         # wrap
         wrap(self)
 
+        # Checks when the distance of the shot is almost over
+        if self.distance_traveled > CONFIG['PLAYER_SHOT_FADE_START']:
```
Comment (tobiasbp) [inline my_game.py:222]: Please use formula: _range - variable_

File: my_game.toml [hunk 5]
```diff
@@ -17,6 +17,8 @@ PLAYER_SPEED_LIMIT = 5  # px/update
 PLAYER_INVINCIBILITY_SECONDS = 5  # secs
 PLAYER_FIRE_RATE = 0.2  # secs
 PLAYER_SHOT_RANGE = 400  # px
+PLAYER_SHOT_FADE_SPEED = 0.95
+PLAYER_SHOT_FADE_START = 300
```
Comment (tobiasbp) [inline my_game.toml:21]: Describe what the variables do.

**Review by tobiasbp — CHANGES_REQUESTED** [review]

File: my_game.py [hunk 6]
```diff
@@ -218,6 +218,11 @@ def update(self):
         # wrap
         wrap(self)
 
+        # Checks when the distance of the shot is almost over
+        if self.distance_traveled > CONFIG['PLAYER_SHOT_RANGE'] - CONFIG['PLAYER_SHOT_FADE_START']:
+            # 5% of the alpha every frame.
```
Comment (tobiasbp) [inline my_game.py]: We don't know that it's 5?

**Review by tobiasbp — APPROVED** [review]

**Pull request closed without merging**